File: ruby/day1.rb
def input
  File.read('../inputs/day1.txt')
end

def part1
  calories = input.split("\n\n").map { |foods| foods.split("\n").map(&:to_i).sum }
  calories.max
end

def part2
  calories = input.split("\n\n").map { |foods| foods.split("\n").map(&:to_i).sum }.sort.reverse
  calories[0..2].sum
end

puts part1
puts part2

File: rust/src/bin/day1.rs
use std::fs;

fn main() {
    println!("{}", part1());
    println!("{}", part2());
}

fn part1() -> i32 {
    input()
        .split("\n\n")
        .map(|foods| {
            foods
                .split("\n")
                .map(|val| val.parse::<i32>().unwrap())
                .sum()
        })
        .max()
        .unwrap()
}

fn part2() -> i32 {
    let mut calories: Vec<i32> = input()
        .split("\n\n")
        .map(|foods| {
            foods
                .split("\n")
                .map(|val| val.parse::<i32>().unwrap())
                .sum()
        })
        .collect();
    calories.sort();
    calories.reverse();
    calories.into_iter().take(3).sum()
}

fn input() -> String {
    String::from_utf8(fs::read("../inputs/day1.txt").unwrap()).unwrap()
}
